### PR TITLE
Fix deprecations with symfony 3.4 and enable sf 4 support

### LINKED ISSUE #1316: Symfony 4 support
Hello, 
While trying to install FOSElastica, I have this error. 
```
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Installation request for friendsofsymfony/elastica-bundle dev-master -> satisfiable by friendsofsymfony/elastica-bundle[dev-master].
    - friendsofsymfony/elastica-bundle dev-master requires symfony/framework-bundle ^2.7 || ^3.1 -> satisfiable by symfony/framework-bundle[2.7.x-dev, 2.8.x-dev, 3.1.x-dev, 3.2.x-dev, 3.3.x-dev, 3.4.x-dev, v2.7.0, v2.7.0-BETA1, v2.7.0-BETA2, v2.7.1, v2.7.10, v2.7.11,
v2.7.12, v2.7.13, v2.7.14, v2.7.15, v2.7.16, v2.7.17, v2.7.18, v2.7.19, v2.7.2, v2.7.20, v2.7.21, v2.7.22, v2.7.23, v2.7.24, v2.7.25, v2.7.26, v2.7.27, v2.7.28, v2.7.29, v2.7.3, v2.7.30, v2.7.31, v2.7.32, v2.7.33, v2.7.34, v2.7.35, v2.7.36, v2.7.37, v2.7.38, v2.7.4, v
2.7.5, v2.7.6, v2.7.7, v2.7.8, v2.7.9, v2.8.0, v2.8.0-BETA1, v2.8.1, v2.8.10, v2.8.11, v2.8.12, v2.8.13, v2.8.14, v2.8.15, v2.8.16, v2.8.17, v2.8.18, v2.8.19, v2.8.2, v2.8.20, v2.8.21, v2.8.22, v2.8.23, v2.8.24, v2.8.25, v2.8.26, v2.8.27, v2.8.28, v2.8.29, v2.8.3, v2.
8.30, v2.8.31, v2.8.4, v2.8.5, v2.8.6, v2.8.7, v2.8.8, v2.8.9, v3.1.0, v3.1.0-BETA1, v3.1.0-RC1, v3.1.1, v3.1.10, v3.1.2, v3.1.3, v3.1.4, v3.1.5, v3.1.6, v3.1.7, v3.1.8, v3.1.9, v3.2.0, v3.2.0-BETA1, v3.2.0-RC1, v3.2.0-RC2, v3.2.1, v3.2.10, v3.2.11, v3.2.12, v3.2.13,
v3.2.14, v3.2.2, v3.2.3, v3.2.4, v3.2.5, v3.2.6, v3.2.7, v3.2.8, v3.2.9, v3.3.0, v3.3.0-BETA1, v3.3.0-RC1, v3.3.1, v3.3.10, v3.3.11, v3.3.12, v3.3.13, v3.3.2, v3.3.3, v3.3.4, v3.3.5, v3.3.6, v3.3.7, v3.3.8, v3.3.9, v3.4.0, v3.4.0-BETA1, v3.4.0-BETA2, v3.4.0-BETA3, v3.
4.0-BETA4, v3.4.0-RC1, v3.4.0-RC2] but these conflict with your requirements or minimum-stability.


```
Is Symfony4 compatible with FosElastica ?

### PR DESCRIPTION
**Pull request by Tobion**
Based on #1326 and replaces #1302, #1315 and fixes #1316

### CODE REVIEW

**Comment by XWB** [comment]
Looks good. Just a small rebase with the master branch is required.

**Comment by Tobion** [comment]
@makasim are you gonna fix the conflicts here as well?

**Comment by makasim** [comment]
@Tobion definitely not today. Is it done?

**Comment by Tobion** [comment]
Well it was done. But you changed alot of stuff like changing travis, dropping support for symfony 2.7 without review and without changing a changelog entry. So there are several inconsistencies in the changelog for example.

**Comment by makasim** [comment]
@Tobion there is a checkbox "Allow edits from editors". Could you please set it on so that I can push into this PR directly?

**Comment by Tobion** [comment]
It is set aleady.

**Comment by makasim** [comment]
I've resolved conflicts though I haven't tested it yet. Let's see what Travis says.

**Comment by XWB** [comment]
> 1) FOS\ElasticaBundle\Tests\Functional\ClientTest::testContainerSource
Symfony\Component\DependencyInjection\Exception\ServiceNotFoundException: The service "fos_elastica.command.populate" has a dependency on a non-existent service "fos_elastica.pager_perister_registry".

File: .travis.yml [hunk 1]
```diff
@@ -6,43 +6,57 @@ cache:
...
+    - php: 7.2
+      env: COVERAGE=true PHPUNIT_FLAGS="-v --coverage-clover=coverage.clover"
+
+    # Install all SF components in the same major version, see https://github.com/dunglas/symfony-lock
+    - php: 7.2
+      env: SYMFONY_VERSION="^2"
+    - php: 7.2
+      env: SYMFONY_VERSION="^3"
+    - php: 7.2
+      env: SYMFONY_VERSION=^4"
```
Comment (stloyd) [inline .travis.yml]: Missing quote.

File: Command/SearchCommand.php [hunk 2]
```diff
@@ -43,14 +52,10 @@ protected function configure()
...
 
-    /**
-     * {@inheritdoc}
-     */
     protected function execute(InputInterface $input, OutputInterface $output)
     {
         $indexName = $input->getOption('index');
-        /** @var $index \Elastica\Index */
-        $index = $this->getContainer()->get('fos_elastica.index_manager')->getIndex($indexName ? $indexName : null);
+        $index = $this->indexManager->getIndex($indexName ? $indexName : null);
```
Comment (stloyd) [inline Command/SearchCommand.php]: `$indexName ?: null`

File: composer.json [hunk 3]
```diff
@@ -28,13 +28,13 @@
         "doctrine/phpcr-odm": "^1",
         "jackalope/jackalope-doctrine-dbal": "^1.1",
         "jms/serializer-bundle": "^2.0",
-        "phpunit/phpunit": "^4.8 || ^5.0",
+        "phpunit/phpunit": "^^5.0",
```
Comment (stloyd) [inline composer.json]: Doubled `^`

**Comment by makasim** [comment]
It seems like we cannot support 2.x, 3.x and 4.x at the same time [because of](https://travis-ci.org/FriendsOfSymfony/FOSElasticaBundle/jobs/315889569)

```
1) FOS\ElasticaBundle\Tests\DependencyInjection\FOSElasticaExtensionTest::testShouldAddParentParamToObjectPersisterCall
Error: Class 'Symfony\Component\DependencyInjection\DefinitionDecorator' not found
```

@Tobion @XWB Dont you mind droping support of Symfony 2.x ?

File: .travis.yml [hunk 4]
```diff
@@ -6,43 +6,57 @@ cache:
...
+    - php: 7.2
+      env: COVERAGE=true PHPUNIT_FLAGS="-v --coverage-clover=coverage.clover"
+
+    # Install all SF components in the same major version, see https://github.com/dunglas/symfony-lock
+    - php: 7.2
+      env: SYMFONY_VERSION="^2"
+    - php: 7.2
+      env: SYMFONY_VERSION="^3"
+    - php: 7.2
+      env: SYMFONY_VERSION="^4"
```
Comment (Tobion) [inline .travis.yml]: that does not seem useful as it is basically the same as the `php: 7.2 env: COVERAGE=true` line

**Comment by Tobion** [comment]
I'm ok to drop symfon 2.x support. But why also drop sf 3.2 and 3.3 support? Minmum of 3.4 or 4.0 which just got released, seems too much.
I'd rather drop php 5.6 support and update to phpunit 6.

**Comment by makasim** [comment]
@Tobion I am fine with lowering it a little bit

**Comment by Tobion** [comment]
@makasim Finally, tests pass. This is ready. Also added autowiring support.

**Comment by MJBGO** [comment]
When will this be merged ? Thanks :)

**Comment by makasim** [comment]
@Tobion thank you!

**Pull request merged**